@xen-orchestra/lite/src/router/index.ts

```diff
@@ -25,6 +28,14 @@ const router = createRouter({
           component: HostDashboardView,
         },
       ],
+      beforeEnter: (to, _, next) => {
+        const hostStore = useHostStore();
+        const host = hostStore.getRecordByUuid(to.params["uuid"] as string);
+        if (host === undefined) {
+          next({ name: "notFound" });
+        }
+        next();
```

Comment:
```suggestion
          if (host === undefined) next({ name: "notFound" });
          else next();
```
See [doc](https://router.vuejs.org/guide/advanced/navigation-guards.html#optional-third-argument-next)